include/seqan3/range/view/pairwise_combine.hpp

```diff
@@ -0,0 +1,595 @@
+    }
+    //!\}
+
+    /*!\name Capacity
+     * \{
+     */
+
+    /*!\brief Returns the size of the range.
+     *
+     * \details
+     *
+     * The size can only be computed when the entire range models std::ranges::RandomAccessRange.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr size_t size() const noexcept
+    //!\cond
+        requires std::RandomAccessIterator<iterator_type>
+    //!\endcond
+    {
+        return static_cast<size_t>(cend() - cbegin());
+    }
+    //!\}
+
+private:
+
+    //!\brief The underling range.
+    source_range_type src_range{};
+    //!\brief The cached iterator pointing to the last element of the source range.
+    std::ranges::iterator_t<source_range_type> back_iterator{};
+};
+
+/*!\name Type deduction guides
+ * \relates seqan3::detail::pairwise_combine_view
+ * \{
+ */
+
+//!\brief Deduces the source range type from the passed view.
+template <std::ranges::View other_type>
+pairwise_combine_view(other_type view) -> pairwise_combine_view<std::remove_reference_t<other_type>>;
+
+//!\brief Deduces the source range type from the passed range.
+template <std::ranges::Range range_type>
+    requires !std::ranges::View<range_type>
+pairwise_combine_view(range_type & range) -> pairwise_combine_view<decltype(std::declval<range_type &>() | view::all)>;
+//!\}
+
+} // namespace seqan3::detail
+
+namespace seqan3::view
+{
+/*!\name General purpose views
+ * \{
+ */
+
+/*!\brief             A view adapter that generates all pairwise combinations of the elements of the source range.
+ * \tparam urng_t     The type of the range being processed. See below for requirements.
+ * \param[in] urange  The range being processed.
+ * \returns           A range over all pairwise combinations. See below for the properties of the returned range.
+ * \ingroup view
+ *
+ * \details
+ *
+ * This view generates two-element tuples representing all possible combinations of the elements of the source range
+ * while ignoring the order of the elements. If the source range has less than two elements the returned range is empty,
+ * otherwise the size of the returned view corresponds to the binomial coefficient `n choose 2`, where `n` is the
+ * size of the source range. The reference type of this range is a tuple over the reference type of the source range.
+ * Constness will not be propagated to the reference types of the source range.
+ *
+ * ### View properties
+ *
+ * | range concepts and reference_t  | `urng_t` (underlying range type)      | `rrng_t` (returned range type)                                       |
+ * |---------------------------------|:-------------------------------------:|:--------------------------------------------------------------------:|
+ * | std::ranges::InputRange         |                                       | *undefined*                                                          |
```

Comment:
The convention for other views was to write *required* | *preserved* for these cases, too. 
Of course Input is required, if Forward is required. It is debatable whether it is then also guaranteed, but like I said the style for the others is to write the above.

Reply:
IDC